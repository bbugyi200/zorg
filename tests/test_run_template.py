"""Tests for the zorg project's 'template' CLI command."""

from __future__ import annotations

from pathlib import Path

from _pytest.capture import CaptureFixture
from syrupy.assertion import SnapshotAssertion as Snapshot

from . import common as c


def test_template_render_day_log(
    main: c.MainType,
    capsys: CaptureFixture,
    zettel_dir: Path,
    snapshot: Snapshot,
) -> None:
    """Test that the 'day_log.zot' template renders as expected."""
    argv = [
        "--dir",
        str(zettel_dir),
        "template",
        "render",
        "day_log.zot",
        "date=19910304",
    ]
    exit_code = main(*argv)

    assert exit_code == 0
    capture = capsys.readouterr()
    assert capture.out == snapshot


def test_template_render_hello(
    main: c.MainType, capsys: CaptureFixture, zettel_dir: Path
) -> None:
    """Test that the 'hello.zot' template renders as expected."""
    argv = [
        "--dir",
        str(zettel_dir),
        "template",
        "render",
        "hello.zot",
        "first_name=John",
        "last_name=Smith",
    ]
    exit_code = main(*argv)

    assert exit_code == 0
    capture = capsys.readouterr()
    assert capture.out == "Hello John Smith\n"
