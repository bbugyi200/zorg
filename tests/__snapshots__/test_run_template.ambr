# serializer version: 1
# name: test_template_render_day_log[YAMLConfigFile]
  '''
  # 1991-03-04.Mon
  #
  # ^ = [[1991/19910304]]
  # < = [[1991/19910303_day]]
  # > = [[1991/19910305_day]]
  # @ = [[day_log.zot]]
  
  o #gtd pomodoro
    * Google Calendar (bbugyi@google + bryanbugyi34@gmail)
    * Process yesterday's bujo log (e.g. close events, add missing pomodoros, rollover goals)
    * Copy any important gchat convos from yesterday into chat/*.txt
    * Check today's tickler file: [[tick_04]]
    * Fill out yesterday's habit tracker: [[1991/19910303_habit]]
    * go/g3co
    * Review Google Keep Inbox
    * Review [[inbox]]
    * Review '@A' todos (are any >=2d old? which '@B' todos should be promoted?)
    * <50 emails in bryanbugyi34@gmail
    * <100 emails in bbugyi@google
    * Review ~/org diffs from yesterday and commit them to git
  
  ######### @event
  
  ######### @pomodoro
  
  '''
# ---
